avm/res/api-management/service/main.bicep

```diff
@@ -0,0 +1,525 @@
+  Contributor: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'b24988ac-6180-42a0-ab88-20f7382dd24c')
+  Owner: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '8e3af657-a8ff-443c-a75c-2fe8c4bcb635')
+  Reader: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'acdd72a7-3385-48ef-bd42-f606fba81ae7')
+  'Role Based Access Control Administrator (Preview)': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'f58310d9-a9f6-439a-9e8d-f62e7b41a168')
+  'User Access Administrator': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '18d7d88d-d35e-4fb5-a5c3-7773c20a72d9')
+}
+
+resource avmTelemetry 'Microsoft.Resources/deployments@2023-07-01' = if (enableTelemetry) {
+  name: '46d3xbcp.res.apimanagement-service.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name, location), 0, 4)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+resource service 'Microsoft.ApiManagement/service@2021-08-01' = {
+  name: name
+  location: location
+  tags: tags
+  sku: {
+    name: sku
+    capacity: skuCount
+  }
+  zones: zones
+  identity: identity
+  properties: {
+    publisherEmail: publisherEmail
+    publisherName: publisherName
+    notificationSenderEmail: notificationSenderEmail
+    hostnameConfigurations: hostnameConfigurations
+    additionalLocations: additionalLocations
+    customProperties: customProperties
+    certificates: certificates
+    enableClientCertificate: enableClientCertificate ? true : null
+    disableGateway: disableGateway
+    virtualNetworkType: virtualNetworkType
+    virtualNetworkConfiguration: !empty(subnetResourceId) ? {
+      subnetResourceId: subnetResourceId
+    } : null
+    apiVersionConstraint: !empty(minApiVersion) ? {
+      minApiVersion: minApiVersion
+    } : null
+    restore: restore
+  }
+}
+
+module service_apis 'api/main.bicep' = [for (api, index) in apis: {
+  name: '${uniqueString(deployment().name, location)}-Apim-Api-${index}'
+  params: {
+    apiManagementServiceName: service.name
+    displayName: api.displayName
+    name: api.name
+    path: api.path
+    apiDescription: api.?apiDescription
+    apiRevision: api.?apiRevision
+    apiRevisionDescription: api.?apiRevisionDescription
+    apiType: api.?apiType
+    apiVersion: api.?apiVersion
+    apiVersionDescription: api.?apiVersionDescription
+    apiVersionSetId: api.?apiVersionSetId
+    authenticationSettings: api.?authenticationSettings
+    format: api.?format ?? 'openapi'
+    isCurrent: api.?isCurrent
+    protocols: api.?protocols
+    policies: api.?policies
```

Comment:
@AlexanderSehr The issue is related to a null being passed into a copy loop within `api/main.bicep` which loops for policies.

Need to check on the behaviour of `length` with null.

Reply:
@krbar @AlexanderSehr Ok. The error is not related to the `policies` parameter in `main.bicep` for the APIM service. It is related to a nested module under `api/main.bicep` which also has a `policies` parameter.

```bicep
param policies array = []
```

As a result, the `for` loop for configuring policies under the API is using a `null` in the `length` function of the expanded template which is failing.

From what I can tell similarly in ARM `length` would fail on a `null` argument. As a result, I think this issue is desired. But I admit we can improve the reporting message in PSRule, which I'll log a card for.

But happy for further feedback/ suggestions or to be proven wrong.

Reply:
@BernieWhite you are right, I came to the same conclusion and updated the implementation of this parameter. Now PSRule shows the results properly and the template also deploys without issues. Thank you for investigating it!